### DISCUSSION TITLE
Welcome to MOOSE Discussions!

### DISCUSSION THREAD

**Discussion by aeslaughter** [idaholab/moose#15774] [Q&A General] [answered]
## Welcome!
We are exploring the use of Discussions as place to connect with other members of the MOOSE community. We hope that you:
  * Ask questions.
  * Share ideas.
  * Engage with other MOOSE users and developers.

  To get started, comment below with an introduction of yourself and tell us about what you doing with MOOSE.

**Comment by aeslaughter** [comment]
We are just testing this out, but I believe this may be a replacement (or at least an alternative) to the mailing list.

**Comment by somu15** [comment]
I am Som. My background is in uncertainty quantification. I am currently developing a fluid-structure interaction module in MOOSE.

**Comment by loganharbour** [comment]
@aeslaughter, where's your intro?

**Comment by aeslaughter** [comment]
I am Andrew and have been a MOOSE developer since 2013. My academic background is in engineering mechanics. Within MOOSE I have all sorts of jobs, including leading the [stochastic tools](https://www.mooseframework.org/modules/stochastic_tools) module development.

**Comment by loganharbour** [comment]
I've been working with the MOOSE team for about a year now, and have been using MOOSE for the last four years. My primary research interest is radiation transport, but I've also been working recently on general framework development as well.

I'm gonna play around with this answered thing real quick...

**Reply by aeslaughter** [reply]
I am replying, to see what that looks like.

**Reply by loganharbour** [reply]
Hello there!

**Reply by loganharbour** [reply]
...man I really like the threading versus doing this over a standard e-mail list.

**Reply by lindsayad** [reply]
The ability to easily reference discussion numbers in issues and PRs and visa versa seems amazing. I could easily see this replacing the mailing list

**Reply by fdkong** [reply]
How hard to search for a topic on the platform?

**Comment by permcody** [comment]
Hi, I'm Cody and I'm currently managing the Computational Frameworks team at the Idaho National Laboratory who maintain MOOSE. I'm always looking for new collaborative opportunities to get this framework into the community.

**Comment by cticenhour** [comment]
Hi, I'm Casey! I've been working with the MOOSE team for about 2.5 years now, and have been using MOOSE for around the last 4-5 years. My research is focused on electromagnetics and low temperature plasma physics modeling and simulation (with some fusion work here and there). I also do occasional MOOSE framework and module development.

**Comment by aeslaughter** [comment, marked as answer]
I locked the discussion, but "Collaborators" still have the power.

**Reply by loganharbour** [reply]
![200px-Drevil_million_dollars](https://user-images.githubusercontent.com/22062133/90698159-511da080-e23d-11ea-9f74-7ae178b29fad.jpg)

**Reply by rwcarlsen** [reply]
What exactly does discussion locking do?  Does it prevent everyone from adding new posts.  Does it prevent replies?  Does it only affect non-collaborators? etc.

**Reply by cticenhour** [reply]
Locking prevents everyone who doesn't have write access to MOOSE from making new comments.

**Comment by aeslaughter** [comment]
Should we announce this forum on moose-users?

**Reply by loganharbour** [reply]
I’d say post that we’re considering using it more and welcome questions here as well. With a note that we’re not switching yet, we would just like some use here so we can gauge if we like it or not.

**Reply by cticenhour** [reply]
I agree - also, the more feedback we can send up based on user engagement to possibly improve it more to our liking, the better

**Reply by rwcarlsen** [reply]
I'm just not sure how many simultaneous places I can pay attention.  groups, slack, teams, email, github discussions, tiny-town (or whatever), etc.

**Reply by aeslaughter** [reply]
I hope this replaces the mailing list and helps with this problem. I am using github.com/notifications which would include posts to the discussions. If we switch it would actually eliminate one place you need to pay attention to.

**Reply by rwcarlsen** [reply]
I agree.  I know we had contemplated a discourse/discord sort of thing also.  I guess I'm just advocating for keeping the intermediate timeframe where we simultaneously have multiple platforms being used down to a minimum (i.e. short evaluation/transition period).  This would indeed be an improvement - I often forget about the mailing list because I'm just busy other places like github.

**Reply by dschwen** [reply]
I hope we're not hemorrhaging too many users with the switch. Also this support channel requires users to sign up for a new service (whereas pretty much everyone already has email). 
I'm curious as to what you think the advantages of this over email are? I like the "accepted answer"  feature, although it is not always applicable in our threads, where answers from multiple people can be complementary or incremental.

**Reply by rwcarlsen** [reply]
I suspect integration with github issues/PRs will also be nice.  Plus many of the devs are already spending time perusing github - it will be easier for us (at least for me) to engage more regularly I think.  But - needing a github account is a pretty big downside :-/

**Reply by aeslaughter** [reply]
I think the threaded conversation is the main advantage with accepted answers being another nice feature. But, I also like the idea of having all MOOSE activity in one location. It will be easier to link to issues, etc.

**Reply by dschwen** [reply]
No offline support either.

**Reply by loganharbour** [reply]
Is that really an issue?

**Reply by dschwen** [reply]
It's just one more thing that contributed to all those newfangled wannabe email replacements to miserably fail.

**Reply by fdkong** [reply]
`Discussions` have a lot of nice features.  I like it in general. The only issue is that we need to force usrs to create a Github account.

**Reply by aeslaughter** [reply]
How many people are working on MOOSE or MOOSE-based animal don't have GitHub account? We ask people on the mailing list to create issues, post their code to github so we can look at etc. If that is the only reason for not switching, I don't think it is compelling enough.

**Reply by YaqiWang** [reply]
Github allows https pull, why do not they allow anonymous discussions? At least give each repo a control if allowing that or not.

**Reply by lindsayad** [reply]
While I do think there's a danger of losing some users in discussions if we move from the mailing list to this, I also think there's a real chance that moving to this forum would encourage more issues and pull requests (which I guess is a pro or a con depending on how you look at it haha)

**Reply by lindsayad** [reply]
Also, perhaps unlikely, but maybe a greater chance that users put their code on github... It would be pretty awesome to link directly to issues/PRs on their own repository from a MOOSE discussion

**Comment by aeslaughter** [comment]
Testing email reply.

**Comment by fdkong** [comment]
Hello, I am Fande

**Comment by rwcarlsen** [comment]
But can it automatically render links in email replies like
https://www.cheese.com

**Reply by rwcarlsen** [reply]
nice

**Reply by lindsayad** [reply]
It's nice that you have the list-like operation of being able to "unsubscribe". Although I'm sure it's only for the current discussion. In my gmail:
```
But can it automatically render links in email replies like
https://www.cheese.com
> <https://github.com/idaholab/moose/discussions/15774#discussioncomment-83814>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAKJ4STRQRY22GWRDFNSAY3SHJDYHANCNFSM4QFNG7IA>
```

**Comment by lingzou** [comment]
I am Ling Zou; has been and still is heavily involved with many MOOSE-based applications, both code development and simulations.

**Comment by vishankkumar** [comment]
Dear All,

I am new to using moose and would like to know how can I generate a spherical mesh to simulate heat/mass transfer across a spherical particle in 2d or 3d.

Moreover, is the syntax page (https://mooseframework.inl.gov/syntax/index.html) is the best place to look for usage details or there is any other suggestion?

Thanks for your help and suggestion in advance.

Kind regards,
Vishank

**Reply by GiudGiud** [reply]
Hi Vishank, please open another discussions for your particular issue. This discussion was meant to test the Discussions functionality / for people to introduce themselves. If we reply to you here, it will be difficult for others to search for this information later. 

I believe @friedmud has worked on a spherical mesh lately, he may be able to help you.

**Reply by dschwen** [reply]
I wrote `SphereSurfaceMeshGenerator` a while ago, but that is not for solid _balls_, but _spheres_ (which are the boundaries of balls!)

**Reply by WilkAndy** [reply]
Next year i might have to perform a simulation of biological species spread over the entire globe.  Then i'll need a $S^{2}$ mesh.   What fun!

**Comment by johncarter489** [comment]
Hello fellow MOOSE'rs - tis' the Season!   Have MOOSE installed on my linux system and kicked the tires on the examples.  I am looking for a good CFD app.  Viscous flow over curvilinear surfaces and solving for Bernoulli variables.   Also something that can handle longitudinal wave propagation in a viscous fluid would be helpful.  Thank you!  John

**Reply by lindsayad** [reply]
There is soon to be a finite volume implementation of incompressible Navier Stokes (#15644) as far as CFD goes. We will be adding a compressible set of equations implemented with finite volume soonish as well as some turbulence modeling. 

With respect to Bernoulli variables...are you talking about Bernoulli random variables or the Bernoulli equation?

**Comment by avtarsinghh1991** [comment]
Hello everyone

My name is Avtar Singh, a postdoctoral associate at MIT. My research interest is the multiphysics modeling of heterogeneous material. Currently, I am developing a coupled chemo-mechanical modeling framework for solid-state lithium-ion batteries using MOOSE. Earlier, I was using in-house FEM-based Fortran code which is integrated with PETSc (in my Ph.D.). Recently, I have been shifted to MOOSE (after joining MIT). Thank you.

**Comment by saqibhayat556** [comment]
Hello there!
My name is Saqib and I'm new on MOOSE platform. My work is regarding simulations of MSR. Can anybody tell me where I should start from to learn MOOSE. Any help would be appreciated. 
Thanks!

**Reply by aeslaughter** [reply]
https://mooseframework.inl.gov/getting_started/examples_and_tutorials/index.html
